pkg/agent/controller/egress/egress_controller.go

```diff
@@ -209,6 +238,21 @@ func (c *EgressController) onLocalIPUpdate(ip string, added bool) {
 	}
 }
 
+// enqueueEgressesByExternalIPPool enqueues the Egress objects with specific ExternalIPPool,
+// the ExternalIPPool is affected by a Node update/create/delete event or
+// Node leaves/join cluster event or ExternalIPPool changed.
+func (c *EgressController) enqueueEgressesByExternalIPPool(eipName string) {
+	egresses, _ := c.egressLister.List(labels.Everything())
+	count := 0
+	for _, egress := range egresses {
+		if egress.Spec.ExternalIPPool == eipName {
```

Comment:
@tnqn is there a reason why we are not indexing the Egress resources by pool name?

Reply:
No, it's a good suggestion. We'd better to index pool name.

Reply:
done